### PR TITLE
TSK-972 initial commit - add task routing SPI to taskana

### PR DESCRIPTION
**Pull request by BerndBreier**

### CODE REVIEW

File: lib/taskana-core/src/test/java/acceptance/taskrouting/TaskRoutingAccTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
+package acceptance.taskrouting;
+
+import static org.junit.Assert.assertEquals;
```
Comment (benjamineckstein) [inline lib/taskana-core/src/test/java/acceptance/taskrouting/TaskRoutingAccTest.java]: please use junit 5 assertions. junit 4 is no longer available with my PR

File: lib/taskana-core/src/main/java/pro/taskana/taskrouting/TaskRoutingProducer.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                 // route to all task routers
+                 // collect in a set to see whether different workbasket ids are returned
+                 Set<String> workbasketIds = theTaskRouters.stream()
+                               .map(rtr -> rtr.routeToWorkbasketId(task))
+                               .filter(Objects::nonNull)
+                               .collect(Collectors.toSet());
+                 if (workbasketIds.isEmpty()) {
+                     LOGGER.error("No TaskRouter determined a workbasket for task {}.", task);
+                 } else if (workbasketIds.size() > 1) {
+                     LOGGER.error("The TaskRouters determined more than one workbasket for task{}", task);
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/taskrouting/TaskRoutingProducer.java]: whitespace missing for task{}

File: lib/taskana-core/src/test/java/acceptance/taskrouting/TaskRoutingAccTest.java (hunk 1 above)
Comment (BerndBreier) [reply lib/taskana-core/src/test/java/acceptance/taskrouting/TaskRoutingAccTest.java]: Done

File: lib/taskana-core/src/main/java/pro/taskana/taskrouting/TaskRoutingProducer.java (hunk 2 above)
Comment (BerndBreier) [reply lib/taskana-core/src/main/java/pro/taskana/taskrouting/TaskRoutingProducer.java]: done

**Pull request merged**